lib/internal/child_process/serialization.js

```diff
@@ -51,29 +52,47 @@ class ChildProcessDeserializer extends v8.DefaultDeserializer {
 //   (aka 'advanced')
 const advanced = {
   initMessageChannel(channel) {
-    channel[kMessageBuffer] = Buffer.alloc(0);
+    channel[kMessageBuffer] = [];
+    channel[kMessageBufferSize] = 0;
     channel.buffering = false;
   },
 
   *parseChannelMessages(channel, readData) {
     if (readData.length === 0) return;
 
-    let messageBuffer = Buffer.concat([channel[kMessageBuffer], readData]);
-    while (messageBuffer.length > 4) {
-      const size = messageBuffer.readUInt32BE();
-      if (messageBuffer.length < 4 + size) {
-        break;
-      }
+    channel[kMessageBuffer].push(readData);
+    channel[kMessageBufferSize] += readData.length;
+
+    // Index 0 should always be present because we just pushed data into it.
+    let messageBufferHead = channel[kMessageBuffer][0];
+    while (messageBufferHead.length >= 4) {
+      // We call `readUInt32BE` manually here, because this is faster than first converting
+      // it to a buffer and using `readUInt32BE` on that.
+      const fullMessageSize = Buffer.prototype.readUInt32BE.call(messageBufferHead, 0) + 4;
+
+      if (channel[kMessageBufferSize] < fullMessageSize) break;
+
+      const concatenatedBuffer = channel[kMessageBuffer].length === 1 ?
```

Comment:
```suggestion
      const concatenatedBuffer = ArrayLength(channel[kMessageBuffer]) === 1 ?
```

Reply:
`ArrayLength` doesn't exist.

Reply:
I assumed it existed since Primordials.d.ts includes the following line: 'export const ArrayLength: typeof Array.length'. Isn't this the case?

Reply:
You're right, it exists. But it's not a function. It's just the value `1`.